### PR TITLE
overhaul widgets.ANSIText

### PR DESCRIPTION
**Pull request by pazz**
This re-implements the ANSI escape sequence parsing in `alot.widgets.ANSIText`
and is intended to avoid ugly try/catch constructions

see issue #1502

### CODE REVIEW

**Comment by pazz** [comment]
Thanks for the input, @pacien. Try again with the updated patch.
It was actually not a bug but a feature :) I simply did not yet include any 8-bit colours that were set "in one go", several attributes in one escape sequence. Your email now looks as expected for me.
Thanks also for the colordiff tip, that's awesome!

File: alot/widgets/ansi.py [hunk 1]
```diff
@@ -60,60 +64,79 @@ def parse_escapes_to_urwid(text, default_attr=None, default_attr_focus=None,
...
+
+    def reset_attr():
+        attr.update(fg=default_attr.foreground,
+                    bg=default_attr.background, bold=default_attr.bold,
+                    underline=default_attr.underline,
+                    standout=default_attr.underline)
+
+    def update_attr(m):
+        # parameter, intermediate, final bytes in the esc seq
+        pb, ib, fb, = m.groups()
```
Comment (pazz) [inline alot/widgets/ansi.py:111]: Codacy Issue found: [Unused variable 'ib'](https://app.codacy.com/manual/patricktotzke/alot/pullRequest?prid=5492258)

**Comment by pacien** [comment]
I confirm that the new patch now allows colours to be displayed, however the bold formatting doesn't seem to be applied on the two coloured lines in the example above.

**Comment by pazz** [comment]
> I confirm that the new patch now allows colours to be displayed, however the bold formatting doesn't seem to be applied on the two coloured lines in the example above.

Thanks. Try once more :)

**Pull request merged**